### ISSUE TITLE
RecordArray doesn't update when new records are loaded

### ISSUE THREAD

**Issue by darthdeus** [emberjs/data#751]
tl;dr: If you have a `RecordArray` or `FilteredRecordArray` and bind to `lastObject`, loading new record won't update the binding, even though observer fires. It only updates once per two records added

[Here's a JSBin](http://jsbin.com/ogorab/148/edit)

``` javascript
allPosts: function() {
  return App.Post.all();
}.property(),

lastPost: function() {
  return this.get("allPosts.lastObject");
}.property("allPosts.length"),
```

It doesn't matter if the new record is added via `App.Post.createRecord()`, or loaded via `App.Post.find(3)`, it still only updates when two records are added. Both with FixtureAdapter and RESTAdapter.

Observers do fire as the JSBin shows, but the bound property doesn't change, even if you try something like `this.notifyPropertyChange("lastPost")` from the observer.

The `lastPost` property also gets triggered every time, the problem is that `lastObject` doesn't return the last object whenever a new record is loaded.
